optimize/what-is-optimize.md

```diff
@@ -0,0 +1,33 @@
+---
+id: what-is-optimize
+title: What is Optimize?
+description: "Leverage process data and analyze areas for improvement."
+---
+
+:::note
+New to Optimize? Visit our introductory guide to [Optimize](/docs/next/guides/improve-processes-with-optimize/) to get started.
+:::
+
+Camunda Platform 8 is built to handle three key aspects of process automation:
+
+- Design
+- Automate
+- Improve
+
+A user can design process flows through our [Modeler](/docs/next/components/modeler/about-modeler/). In a production scenario, you can deploy through Desktop Modeler, Web Modeler, or programmatically. A user can use [Tasklist](/docs/next/components/tasklist/introduction-to-tasklist/) to review and complete tasks, and [Operate](/docs/next/components/operate/operate-introduction) to view and analyze process instances.
```

Comment:
👋 👀 

FYI I'm going to add an issue for the outcome of this very long note, but I also want to get feedback from people @christinaausley @akeller @korthout in case you think I'm wasting my time and being overly dramatic. 

This is ....kind of sucky. These links all had to change [from the originals](https://github.com/camunda/camunda-platform-docs/blob/bdf1dccec35bcad8aeb95d1f17d47c4f60b4c75c/docs/components/optimize/what-is-optimize.md?plain=1#L17). e.g. `[Operate](/components/operate/operate-introduction.md)` changed to `[Operate](/docs/next/components/operate/operate-introduction)`

## Why?

This is because the links cross from one docs plugin to the other (that's how we implement multi-versioning in docusaurus -- two docs plugins, one for each set of versions). 

Previously we were able to specify the doc ID and path via source....and docusaurus would magically find the doc you were linking to. We didn't need to know the URL, and we didn't have to specify a version because it used the same version you were browsing.

But now that they're cross-plugin, we have to specify the **full URL** of the target doc. This includes the version....which to some degree makes sense, because d-rex doesn't know that Optimize v3.8.0 is associated with C8 v8.0. 

## Why it's awful

Where it especially stinks though, is in the "next" and "current" docs. In the "next" docs, we'll want Optimize to link to C8 "next", and C8 to link to Optimize "next". When we cut versions 3.9.0/8.1 from that, we'll have to go through and replace every markdown link in those versions to remove the "/next" from the path. And then we'll have to go through the current 3.8.0/8.0 docs, and replace all _those_ markdown links to add "/3.8.0" or "8.0".

This feels like **a lot** of friction. I didn't catch this when building the POCs. It'll be a pain for people writing docs because they can't just copy/paste their changes from the next version into current (or vice versa). It'll be a pain for reviewers because we'll overlook it all the time. It'll be a pain when we cut new versions. 

There is also friction caused by sidebars needing to be expanded to include full URLs when crossing docs plugins. But I'm comfortable over-looking that friction because we don't modify the sidebars files very often. 

Introducing friction on every markdown link, though, seems like it'll be extremely un-fun.

## What I think we can do about it

So right now my theory is this: we could write [an MDX plugin](https://docusaurus.io/docs/markdown-features/plugins) for these docs so that we can write something like this in the markdown: 

```md
Hello I like Camunda especially [the modeler]($docs$/components/modeler/about-modeler)
```

and the mdx plugin would see the `$docs$` token, and replace it with the correctly mapped version path. e.g:

- if it was found in a doc from optimize/next, it would replace it with `/docs/next/`
- if it was found in a doc from optimize/3.7.0, it would replace it with `/docs/1.3/`
- if it was found in a doc from optimize/3.8.0, `/docs/` (at least right now....when 8.0 is current.)
- and same to go backwards, from links found in docs from c8 into optimize. 

## A caveat

I haven't worked with MDX plugins before. I'm looking forward to the fun of writing one, but I don't really know what is or isn't possible in the context of a docusaurus site. So I guess we'll find out if my theory (or something similar) is anywhere feasible.

Reply:
two bits of context I should have shared, from a conversation with @RomanJRW: 

1. [This list of optimize releases](https://confluence.camunda.com/display/CO/Optimize+Release+Overview). 
2. Optimize has decided to align their release schedule with that of C8. (I don't have this in writing anywhere, but I think that's what Josh told me)

Those two facts, combined with the fact that we only have "Optimize" docs starting in the platform version 1.3 docs, lead me to believe that the versioning _is_ tightly coupled, with the specific versions you listed, and will be in the future.

Reply:
@pepopowitz As part of the summerhackdays 2022, I recently created some MDX pages for Docusaurus that embed 2 custom build React components. This was rather simple TBH. I was surprised by the freedom we had in there. 

I think that MDX plugins work similarly, but are embedded in a different way. As I understand it, the plugins are configured in docusaurus.config.js, instead of imported and embedded in each markdown file directly.

Reply:
🔗 See the other comment in this file for an example of escaping this linking hell!

Reply:
For posterity: it sounds like docusaurus has plans for v3 to [enable different plugins to see each other's stuff](https://github.com/facebook/docusaurus/issues/6370). When that happens we might be able to undo this work 🙌